### ISSUE TITLE
Dictionary of [Text, Interface "..."] triggers false-positive LC0003 and LC0005 errors: interface name mistaken for Object ID / casing violation

### ISSUE THREAD

**Issue by arsag** [StefanMaron/BusinessCentral.LinterCop#1225]
**1. Description**
Declaring a variable of type Dictionary of [Text, Interface "IMyInterface"] triggers two false-positive linter errors:

LC0003: "Do not use an Object ID for properties or variables declaration. Use Dictionary instead." — the interface name in the value type position is misidentified as an Object ID.
LC0005: "Wrong casing detected! Use Dictionary instead." — a casing violation is reported on the same token.
The code is valid AL and compiles successfully once the warnings are suppressed with #pragma warning disable LC0003, LC0005. No actual Object ID or casing issue exists.

**2. To Reproduce**
Declare a variable using Dictionary of [Text, Interface "IMyInterface"]:

``` AL
codeunit 50000 "My Codeunit"
{
    var
        MyDict: Dictionary of [Text, Interface "IMyInterface"]; // LC0003 + LC0005 reported here
}

interface "IMyInterface"
{
    procedure DoSomething();
}
```

**3. Expected behavior**
No linter errors or warnings. Dictionary of [Text, Interface "..."] is valid AL syntax; the interface name in the value type position should not be parsed as an Object ID or trigger a casing check.

**4. Actual behavior**
Two linter errors are reported on the interface name token inside the Dictionary value type:

> error LC0003: Do not use an Object ID for properties or variables declaration. Use Dictionary instead.
> error LC0005: Wrong casing detected! Use Dictionary instead.

**Comment by Arthurvdv** [comment]
Hi @arsag!

I've did some work on this on the successor of the BusinessCentral.LinterCop and I would invite you to migrate to ALCOps, where these false positives are already solved and we’re continuing development there (see also the note on the [README](https://github.com/StefanMaron/BusinessCentral.LinterCop/blob/master/README.md)).

A starting point can be the [Migrating from BusinessCentral.LinterCop](https://alcops.dev/docs/lintercop-migration/) on the alcops.dev website. Also on the documentation are the setup for VS Code and integration with pipelines for GitHub and/Azure DevOps.

I'm going to close the issue here, but if you have any questions regarding ALCops, it would be helpful to create a discussion/issue on one of the repos of https://github.com/ALCops
